src/services/refactors/inlineVariable.ts

```diff
@@ -112,7 +116,12 @@ registerRefactor(refactorName, {
         const { references, declaration, replacement } = info;
         const edits = textChanges.ChangeTracker.with(context, tracker => {
             for (const node of references) {
-                tracker.replaceNode(file, node, getReplacementExpression(node, replacement));
+                if (isStringLiteral(replacement) && isIdentifier(node) && isTemplateSpan(node.parent)) {

```

Comment:
One edge case I think we should consider are cases like this:

```ts
import { $ } from "execa";

const message = "Hello, World!";

await $`echo ${message}`;
```

In this case, inlining the value would cause a behavior change.

I think this will need to be change to only allow this special refactor when the template span when it isn't a child of the template of a `TaggedTemplateExpression`.

Reply:
nice catch!